core/src/main/java/org/elasticsearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -110,10 +110,8 @@ public void validate(RoutingTableValidation validation, MetaData metaData) {
 
     }
 
-    /**
-     * validate based on a meta data, returning failures found
-     */
-    public List<String> validate(IndexMetaData indexMetaData) {
+    // validate based on a meta data, returning failures found
+    List<String> validate(IndexMetaData indexMetaData) {
```

Comment:
same here. I think we should throw an exception when running into an error. No need to accumulate things and return them